IPython/core/interactiveshell.py

```diff
@@ -2781,6 +2784,15 @@ def run_ast_nodes(self, nodelist, cell_name, interactivity='last_expr',
         if not nodelist:
             return
 
+        if interactivity == 'last_expr_or_assign':
+            if isinstance(nodelist[-1], ast.Assign):
+                asg = nodelist[-1]
+                if len(asg.targets) == 1:
+                    nnode = ast.Expr(ast.Name(asg.targets[0].id, ast.Load()))
```

Comment:
Hey ! ... it's not in the released version ! It's been merged a year ago, why no release? 
![1pns0e](https://cloud.githubusercontent.com/assets/335567/26426646/72f2fc68-408e-11e7-9f97-e76ede5ffbfd.jpg)